File: 167_两数之和 II - 输入有序数组/main.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l = 0
        h = len(numbers) -1
        while l < h:
            if numbers[l] + numbers[h] < target:
                l += 1
            elif numbers[l] + numbers[h] > target:
                h -= 1
            else:
                return [l+1, h+1]
        return None

File: 167_两数之和 II - 输入有序数组/main.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        vector<int> ret; 
        int l = 0, h = numbers.size() -1;
        while( l < h){
            if(numbers[l] + numbers[h] < target)
                ++l;
            else if(numbers[l] + numbers[h] > target)
                --h;
            else{
                ret.push_back(l+1);
                ret.push_back(h+1);
                break;
            }
        }
        return ret;
    }
};